src/urban_disco/grid/core.clj

```diff
@@ -0,0 +1,42 @@
+(ns urban-disco.grid.core
+  (:require [grid.constants :as constants]
+            [grid.grid      :as grid]
+            [grid.tile      :as tile]
+            [grid.group     :as group]
+            [grid.neighbors :as neighbors]
+            [grid.base-grid :as base])
+  (:gen-class))
+
+
+(def base-grid (base/build-grid))
+(def starting-tile (grid/get-tile base-grid 4 2))
+(def goal-tile (grid/get-tile base-grid 0 3))
+
+(defn generate-grid [base-grid start-tile goal-tile]
+  (loop [start      [start-tile]
+         goal       [goal-tile]
+         unexplored (-> base-grid (grid/remove-tile start-tile) (grid/remove-tile goal-tile))
+         final-grid base-grid]
+    (if (not-empty unexplored)
+      (let [rand-start           (neighbors/rand-unexplored final-grid start)
+            rand-goal            (neighbors/rand-unexplored final-grid goal)
+            rand-start-neighbor  (neighbors/rand-unexplored-neighbor final-grid rand-start)
+            rand-goal-neighbor   (neighbors/rand-unexplored-neighbor final-grid rand-goal rand-start-neighbor)]
+        (recur
```

Comment:
Since you defined the `loop` above with 4 bindings, every call to `recur` needs to have 4 arguments. This has 5. It looks like maybe you just need to remove the last arg but I haven't looked closely yet.

Reply:
Oops, I was using a fifth argument to test while writing it. I thought I removed it.